### PR TITLE
Use LWT for task delete statement

### PR DESCRIPTION
**Pull request by cdonati**
Without this, a delete can come in with a timestamp that is earlier than the timestamp of some fields, resulting in a row with only some fields defined.

### CODE REVIEW

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/3030/

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/3039/

File: AppTaskQueue/appscale/taskqueue/queue.py [hunk 1]
```diff
@@ -940,20 +943,25 @@ def _delete_index(self, eta, task_id):
...
       WHERE app = %(app)s AND queue = %(queue)s AND id = %(id)s
-    """)
+      IF EXISTS
+    """, retry_policy=NO_RETRIES)
     parameters = {'app': self.app, 'queue': self.name, 'id': task.id}
-    batch_delete.add(delete_task, parameters=parameters)
+    try:
+      self.db_access.session.execute(delete_task, parameters=parameters)
+    except TRANSIENT_CASSANDRA_ERRORS:
+      retries_left = retries - 1
```
Comment (jeanleonov) [inline AppTaskQueue/appscale/taskqueue/queue.py]: It makes sense to log warning about Transient error and inform that one more attempt will be done.

File: AppTaskQueue/appscale/taskqueue/queue.py [hunk 2]
```diff
@@ -940,20 +943,26 @@ def _delete_index(self, eta, task_id):
...
+      if retries_left <= 0:
+        raise
+      logger.warning('Task deletion may have failed. Retrying.')
```
Comment (jeanleonov) [inline AppTaskQueue/appscale/taskqueue/queue.py]: I'm sorry for being nagging=) but would you mind adding string representation of an error?

**Comment by cdonati** [comment]
@jeanleonov Lightweight transaction. Anytime an 'IF' statement is added to a mutation, it uses a different write path. https://www.datastax.com/dev/blog/lightweight-transactions-in-cassandra-2-0

**Comment by jeanleonov** [comment]
@cdonati , now I see. Quite useful feature

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/3136/
https://ocd.appscale.com:8080/job/Euca-Faststart/999/

**Pull request merged**